Base summoners, their scrolls, interactions, & more

Steven-Baez opened the pull request:
Also adds all the assests & things.

Steven-Baez committed:
when a scroll and a stone are sus

Steven-Baez committed:
lots of work, add scroll's & their enchantments

Steven-Baez committed:
Nekomaster's new texture's! Also start new modid (summonerscrolls)

Steven-Baez committed:
Fully change to "summonerscrolls" modid

Steven-Baez committed:
stuff

Steven-Baez committed:
go on forge a bit?

Steven-Baez committed:
thanks @HellionGames
when the code works :hyperblob:

Steven-Baez committed:
Add item's lang & models

Steven-Baez committed:
Make enchantments scroll specific

Steven-Baez committed:
add loot tables and villager trades

Steven-Baez committed:
add loot tables and villager trades

Steven-Baez committed:
change from weaponsmith to cleric traders

Steven-Baez committed:
try to fix anvil recipes in fabric #2 ?

Steven-Baez committed:
each enchantment spawns its own corresponding mob

Steven-Baez committed:
start zombie summon, add creeper summons & sword summoning

Steven-Baez committed:
Start on summon entities, rendering, and sword summons

Steven-Baez committed:
zombie summon texture

Steven-Baez committed:
bright summon zombie :)

Steven-Baez committed:
start with summon goals & ai

Steven-Baez committed:
fix anvils for fabric, fix #2

Steven-Baez committed:
Working ai! Also recognizes owner, attacks other mobs, and that hurt the owner.

Steven-Baez committed:
non pushable and hittable

Steven-Baez committed:
summon particles

Steven-Baez committed:
lit summon particles

Steven-Baez committed:
weapon cooldown

Steven-Baez committed:
despawn delay

Steven-Baez committed:
invisible summon light

Steven-Baez committed:
summon death sound

Steven-Baez committed:
right click to spawn summon

Steven-Baez committed:
check for right click actions

Steven-Baez committed:
fix summons attacking eachother, half paticles, faster summon

Steven-Baez committed:
ability to remove durability & xp!

Steven-Baez committed:
use summon interface instead of ZombieSummon

Steven-Baez committed:
New textures by Neko!

Steven-Baez committed:
edit zombie summon texture

Steven-Baez committed:
rename summon layer

Steven-Baez committed:
automate some summon mob specific things

Steven-Baez committed:
Add husk summon & all other scrolls!

Steven-Baez committed:
remove necessary scroll ench

Steven-Baez committed:
fix husk renderer

Steven-Baez committed:
fix husk glow layer

Steven-Baez committed:
fix runtime errors

nekomaster1000 committed:
Added missing lang and json entries

Now every texture and displays correctly!

- Fixed inconsistent ordering of lang entries to match the Google Doc
- Changed the Enchantments' suffix from 'Scroll' to 'Summon' (e.g. Zombie Summon)
- Fixed the tab being called 'Summoners Scrolls' mistakenly
- Changed the tab's icon to the Enhancement Scroll

- Added the Enhancement Scroll, which stacks up to 16
- Fixed Husk Summons moving at 0.3 speed (now they move at 0.4 speed)

Steven-Baez committed:
Merge pull request #3 from teamfusion/assetfix

Added missing lang and json entries

Steven-Baez committed:
datagen + cleanup 😎

Steven-Baez committed:
Merge remote-tracking branch 'origin/summonentity' into summonentity

Steven-Baez committed:
scroll xp tooltip + husk on fire

Steven-Baez committed:
fix name issue

Steven-Baez committed:
start work on summon skeleton

Steven-Baez committed:
Clean up, make todos, and datagen!

Steven-Baez committed:
make enchants not appear in the enchantment table

Steven-Baez committed:
Summoner tools and hardcoded skeleton bow stuff 😭

Steven-Baez committed:
bow model

Steven-Baez committed:
oops lang name got changed on accident

Steven-Baez committed:
some more refactoring

Steven-Baez committed:
hold off on parchment for now

Steven-Baez committed:
fix mob type error

Steven-Baez committed:
fix tiny datagen lang error

Steven-Baez committed:
BufferBuilder not started 🙄

Steven-Baez committed:
Add Spider & Stray, Fix Summon Spawning & Attacking

Steven-Baez committed:
large cleanup :)

Steven-Baez committed:
Creeper, Cave Spider, & Fixes!

Steven-Baez committed:
enderman & charged creeper

Steven-Baez committed:
enderman behavior fix

Steven-Baez committed:
Creepers won't explode their owners :)

Steven-Baez committed:
Loot tables - summons now drop scrolls (1% chance)

Steven-Baez committed:
edit the hardcoded drop loot

Dariensg commented on common/src/main/generated/assets/summonerscrolls/lang/en_us.json:
```diff
@@ -0,0 +1,35 @@
+{
+  "block.minecraft.torch": "Invisible Summon Light",
```
Why is this done? This seems not right at all lol

Dariensg commented on common/src/main/generated/assets/summonerscrolls/lang/en_us.json:
```diff
@@ -0,0 +1,35 @@
+  "enchantment.summonerscrolls.zombie_scroll": "Zombie Summon",
+  "entity.summonerscrolls.husk_summon": "Husk Summon",
+  "entity.summonerscrolls.skeleton_summon": "Skeleton Summon",
+  "entity.summonerscrolls.zombie_summon": "Zombie Summon",
```
Seem to be missing several entities

Dariensg commented on common/src/main/generated/data/summonerscrolls/loot_tables/entities/cave_spider_summon.json:
```diff
@@ -0,0 +1,32 @@
+      "functions": [
+        {
+          "function": "minecraft:looting_enchant",
+          "count": 0.01
```
Looting enchant is an int item modifier

Dariensg commented on common/src/main/generated/data/summonerscrolls/loot_tables/entities/charged_creeper_summon.json:
```diff
@@ -0,0 +1,32 @@
+      "functions": [
+        {
+          "function": "minecraft:looting_enchant",
+          "count": 0.01
```
Also an int

Dariensg commented on common/src/main/generated/data/summonerscrolls/loot_tables/entities/creeper_summon.json:
```diff
@@ -0,0 +1,32 @@
+      "functions": [
+        {
+          "function": "minecraft:looting_enchant",
+          "count": 0.01
```
Int

Dariensg commented on common/src/main/generated/data/summonerscrolls/loot_tables/entities/enderman_summon.json:
```diff
@@ -0,0 +1,32 @@
+      "functions": [
+        {
+          "function": "minecraft:looting_enchant",
+          "count": 0.01
```
Int

Dariensg commented on common/src/main/generated/data/summonerscrolls/loot_tables/entities/husk_summon.json:
```diff
@@ -0,0 +1,32 @@
+      "functions": [
+        {
+          "function": "minecraft:looting_enchant",
+          "count": 0.01
```
Int

Dariensg commented on common/src/main/generated/data/summonerscrolls/loot_tables/entities/skeleton_summon.json:
```diff
@@ -0,0 +1,32 @@
+      "functions": [
+        {
+          "function": "minecraft:looting_enchant",
+          "count": 0.01
```
Int

Dariensg commented on common/src/main/generated/data/summonerscrolls/loot_tables/entities/spider_summon.json:
```diff
@@ -0,0 +1,32 @@
+      "functions": [
+        {
+          "function": "minecraft:looting_enchant",
+          "count": 0.01
```
Int

Dariensg commented on common/src/main/generated/data/summonerscrolls/loot_tables/entities/stray_summon.json:
```diff
@@ -0,0 +1,32 @@
+      "functions": [
+        {
+          "function": "minecraft:looting_enchant",
+          "count": 0.01
```
Int

Dariensg commented on common/src/main/generated/data/summonerscrolls/loot_tables/entities/zombie_summon.json:
```diff
@@ -0,0 +1,32 @@
+      "functions": [
+        {
+          "function": "minecraft:looting_enchant",
+          "count": 0.01
```
Int

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/client/render/entity/CaveSpiderSummonRenderer.java:
```diff
@@ -0,0 +1,38 @@
+@Environment(EnvType.CLIENT)
+public class CaveSpiderSummonRenderer<T extends CaveSpiderSummon> extends SpiderSummonRenderer<T> {
+    public static final String SUMMON_NAME = "spider";
+    public static final ResourceLocation SUMMON_LOCATION = new ResourceLocation(SummonerScrolls.MOD_ID, "textures/entity/summon/" + SUMMON_NAME + "_summon.png");
```
No reason to separate the name for usage in this one location. It would make sense if you had a SummonEntityRenderer super class that takes in a summon name in its constructor to do this automatically, but given that it's different for each renderer and only used in making the location, it should be hardcoded.

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/client/render/entity/CreeperSummonRenderer.java:
```diff
@@ -0,0 +1,47 @@
+@Environment(EnvType.CLIENT)
+public class CreeperSummonRenderer extends CreeperRenderer {
+    public static final String SUMMON_NAME = "creeper";
+    public static final ResourceLocation SUMMON_LOCATION = new ResourceLocation(SummonerScrolls.MOD_ID, "textures/entity/summon/" + SUMMON_NAME + "_summon.png");
```
Same as above

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/client/render/entity/EndermanSummonRenderer.java:
```diff
@@ -0,0 +1,47 @@
+@Environment(EnvType.CLIENT)
+public class EndermanSummonRenderer extends EndermanRenderer {
+    public static final String SUMMON_NAME = "enderman";
+    public static final ResourceLocation SUMMON_LOCATION = new ResourceLocation(SummonerScrolls.MOD_ID, "textures/entity/summon/" + SUMMON_NAME + "_summon.png");
```
Again

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/client/render/entity/HuskSummonRenderer.java:
```diff
@@ -0,0 +1,45 @@
+@Environment(EnvType.CLIENT)
+public class HuskSummonRenderer extends HuskRenderer {
+    public static final String SUMMON_NAME = "husk";
+    public static final ResourceLocation SUMMON_LOCATION = new ResourceLocation(SummonerScrolls.MOD_ID, "textures/entity/summon/" + SUMMON_NAME + "_summon.png");
```
Same as above

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/client/render/entity/SkeletonSummonRenderer.java:
```diff
@@ -0,0 +1,55 @@
+@Environment(EnvType.CLIENT)
+public class SkeletonSummonRenderer extends HumanoidMobRenderer<AbstractSkeleton, SkeletonModel<AbstractSkeleton>> {
+    public static final String SUMMON_NAME = "skeleton";
+    public static final ResourceLocation SUMMON_LOCATION = new ResourceLocation(SummonerScrolls.MOD_ID, "textures/entity/summon/" + SUMMON_NAME + "_summon.png");
```
Same as above

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/client/render/entity/SpiderSummonRenderer.java:
```diff
@@ -0,0 +1,54 @@
+@Environment(EnvType.CLIENT)
+public class SpiderSummonRenderer<T extends SpiderSummon> extends SpiderRenderer<T> {
+    public static final String SUMMON_NAME = "spider";
+    public static final ResourceLocation SUMMON_LOCATION = new ResourceLocation(SummonerScrolls.MOD_ID, "textures/entity/summon/" + SUMMON_NAME + "_summon.png");
```
Same as above

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/client/render/entity/ZombieSummonRenderer.java:
```diff
@@ -0,0 +1,45 @@
+@Environment(EnvType.CLIENT)
+public class ZombieSummonRenderer extends ZombieRenderer {
+    public static final String SUMMON_NAME = "zombie";
+    public static final ResourceLocation SUMMON_LOCATION = new ResourceLocation(SummonerScrolls.MOD_ID, "textures/entity/summon/" + SUMMON_NAME + "_summon.png");
```
Same as above

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/BeeSummon.java:
```diff
@@ -0,0 +1,247 @@
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.*;
```
Please avoid wildcard imports

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/BeeSummon.java:
```diff
@@ -0,0 +1,247 @@
+            this.spawnAtLocation(itemstack);
+            this.setItemSlot(EquipmentSlot.OFFHAND, ItemStack.EMPTY);
+        }
+    }
```
Why are you making them drop their lights?

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/BeeSummon.java:
```diff
@@ -0,0 +1,247 @@
+    @Override
+    public void stopBeingAngry() {
+        super.stopBeingAngry();
+    }
```
Why override if you're just calling super?

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/CaveSpiderSummon.java:
```diff
@@ -0,0 +1,67 @@
+import net.minecraft.world.DifficultyInstance;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.effect.MobEffects;
+import net.minecraft.world.entity.*;
```
Again, avoid wildcard

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/CreeperSummon.java:
```diff
@@ -0,0 +1,254 @@
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.*;
```
Avoid wildcard imports

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/EndermanSummon.java:
```diff
@@ -0,0 +1,246 @@
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.*;
```
Avoid wildcard imports

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/EndermanSummon.java:
```diff
@@ -0,0 +1,246 @@
+            this.spawnAtLocation(itemstack);
+            this.setItemSlot(EquipmentSlot.OFFHAND, ItemStack.EMPTY);
+        }
+    }
```
Again, why are you dropping the light?

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/EndermanSummon.java:
```diff
@@ -0,0 +1,246 @@
+    @Override
+    public void stopBeingAngry() {
+        super.stopBeingAngry();
+    }
```
Again, no reason to override if you're just gonna call super.

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/HuskSummon.java:
```diff
@@ -0,0 +1,200 @@
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.effect.MobEffects;
+import net.minecraft.world.entity.*;
```
Avoid wildcard imports

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/HuskSummon.java:
```diff
@@ -0,0 +1,200 @@
+            this.spawnAtLocation(itemstack);
+            this.setItemSlot(EquipmentSlot.OFFHAND, ItemStack.EMPTY);
+        }
+    }
```
Again with dropping the light

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/ISummon.java:
```diff
@@ -0,0 +1,68 @@
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.Mob;
+import net.minecraft.world.entity.PathfinderMob;
+import net.minecraft.world.entity.ai.goal.*;
```
Avoid wildcard imports

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/SkeletonSummon.java:
```diff
@@ -0,0 +1,224 @@
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.*;
```
Avoid wildcards

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/SkeletonSummon.java:
```diff
@@ -0,0 +1,224 @@
+    protected void dropEquipment() {
+        super.dropEquipment();
+        ItemStack itemstack = this.getItemBySlot(EquipmentSlot.OFFHAND);
+        if (!itemstack.isEmpty()) {
+            this.spawnAtLocation(itemstack);
+            this.setItemSlot(EquipmentSlot.OFFHAND, ItemStack.EMPTY);
+        }
```
Why drop the light?

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/SpiderSummon.java:
```diff
@@ -0,0 +1,180 @@
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.*;
```
Avoid wildcards

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/SpiderSummon.java:
```diff
@@ -0,0 +1,180 @@
+    protected void dropEquipment() {
+        super.dropEquipment();
+        ItemStack itemstack = this.getItemBySlot(EquipmentSlot.OFFHAND);
+        if (!itemstack.isEmpty()) {
+            this.spawnAtLocation(itemstack);
+            this.setItemSlot(EquipmentSlot.OFFHAND, ItemStack.EMPTY);
+        }
```
Why drop the light?

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/ZombieSummon.java:
```diff
@@ -0,0 +1,178 @@
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.*;
```
Avoid wildcard

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/ZombieSummon.java:
```diff
@@ -0,0 +1,178 @@
+    protected void dropEquipment() {
+        super.dropEquipment();
+        ItemStack itemstack = this.getItemBySlot(EquipmentSlot.OFFHAND);
+        if (!itemstack.isEmpty()) {
+            this.spawnAtLocation(itemstack);
+            this.setItemSlot(EquipmentSlot.OFFHAND, ItemStack.EMPTY);
+        }
```
Why drop the light?

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/goal/FollowOwnerGoal.java:
```diff
@@ -0,0 +1,138 @@
+
+import java.util.EnumSet;
+
+public class FollowOwnerGoal extends Goal {
```
Doesn't this already exist as a Vanilla goal?

Steven-Baez replied:
It does but since the summons doesn't extend the tameable mob I had to make my own.

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/entity/goal/OwnerHurtByTargetGoal.java:
```diff
@@ -0,0 +1,50 @@
+
+import java.util.EnumSet;
+
+public class OwnerHurtByTargetGoal extends TargetGoal {
```
Doesn't this also already exist as a Vanilla goal? I could be wrong on both of these, but I could've sworn they do, since wolves would use them.

Steven-Baez replied:
Same as above

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/common/item/ScrollItem.java:
```diff
@@ -0,0 +1,38 @@
+    @Override
+    public void appendHoverText(ItemStack stack, @Nullable Level level, List<Component> list, TooltipFlag tooltipFlag) {
+        super.appendHoverText(stack, level, list, tooltipFlag);
+        list.add((new TranslatableComponent("item.summonerscrolls.scroll.xp_warning")).append(" ").append(String.valueOf(ScrollUtil.getScrollXPCount(stack))).withStyle(ChatFormatting.AQUA));
```
What's up with the blank append? And if that's a new line character, please use /n instead of just a new line.

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/mixin/TieredItemMixin.java:
```diff
@@ -0,0 +1,109 @@
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+@Mixin(TieredItem.class)
+public abstract class TieredItemMixin extends Item {
```
Why is this not just an override in the ScrollItem class? Putting it here risks breaking other mods and there's no feasible reason to, given that you wrote the class that this is for in the first place.

Steven-Baez replied:
I couldn't really do that since the scroll item is just like the enchantment book not the actual weapon spawning in the summons.

Dariensg replied:
Ok. I'd try to find a way to do this with an Inject instead though, since Overrides are dangerous. Only one override from any mod can target any given method, so any other mod that tries to override or redirect this same method will be incompatible and I feel like there's a way to do this with an inject.

Dariensg commented on common/src/main/java/com/github/teamfusion/summonerscrolls/mixin/MobMixin.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    @Inject(method = "isSunBurnTick", at = @At("HEAD"), cancellable = true)
+    protected void isSunBurnTick(CallbackInfoReturnable<Boolean> cir) {
```
This seems like it should just be an override in the appropriate summon entity classes. No reason to mixin since you wrote the entity classes yourself.

Dariensg commented on common/src/main/resources/assets/summonerscrolls/models/item/summon_bow.json:
```diff
@@ -0,0 +1,50 @@
+      "scale": [ 0.68, 0.68, 0.68 ]
+    }
+  },
+  "overrides": [
+    {
+      "predicate": {
+        "pulling": 1
+      },
+      "model": "summonerscrolls:item/summon_bow_pulling_0"
+    },
+    {
+      "predicate": {
+        "pulling": 1,
+        "pull": 0.65
+      },
+      "model": "summonerscrolls:item/summon_bow_pulling_1"
+    },
+    {
+      "predicate": {
+        "pulling": 1,
+        "pull": 0.9
+      },
+      "model": "summonerscrolls:item/summon_bow_pulling_2"
+    }
+  ]
```
This could just be because I'm reviewing this over several days, but I don't recall you actually defining this ItemModelPredicate in code anywhere. I'll keep an eye out if it comes up later in the review.

Dariensg commented on fabric/src/main/resources/fabric.mod.json:
```diff
@@ -27,18 +27,20 @@
     "minecraft": ">=1.18.2",
     "fabricloader": ">=0.14",
-    "fabric-api": ">=0.58",
+    "fabric": "*",
```
I would suggest giving a version range, but not entirely necessary

Dariensg commented on quilt/src/main/resources/summonerscrolls.mixins.json:
```diff
@@ -1,7 +1,8 @@
 {
   "required": true,
-  "package": "net.examplemod.mixin.forge",
+  "package": "com.github.teamfusion.summonerscrolls.mixin.fabric",
```
I've never modded Quilt before, so just want to make sure this should be fabric and not Quilt. I know they're built on the same base, so I could still see this being correct.

Steven-Baez replied:
yeah it should be fine, irrc the template comes like this

Dariensg reviewed (changes requested):
Found a few things through here. Mostly minor stuff, but a few things that seem entirely unnecessary to me. First off, most of the common mixins seem like they don't have to be mixins at all, since you wrote the necessary classes yourself. Second, you seem to be dropping the invisible lights in the dropEquipment method for several of the mobs. Not sure why you're doing that. The other big thing was just a lot of wildcard imports, which shouldn't be used according to the Google Java Style Guide (which I urge people to use). Besides that, just minor things here and there. As an overall statement, please don't make PRs this large in the future. PRs should be single, complete features, not entire mods. This becomes dreadful to review when I have to review an entire mod at once instead of one, contained feature.

Steven-Baez committed:
Remove Wildcard Imports

Steven-Baez committed:
Fix Fabric Version Range

Steven-Baez committed:
Fix Looting Int Error

Steven-Baez committed:
Fix Invisible Summon Light Error

Steven-Baez committed:
Fix Missing Names Errors

Steven-Baez committed:
Run datagen for missing lang names

Steven-Baez committed:
Remove Hardcode Summon Names

Steven-Baez committed:
Fix summons dropping lights bug

Steven-Baez committed:
Fix override calling super

Steven-Baez committed:
Fix override calling super

Steven-Baez committed:
Remove sunburntick mixin

Steven-Baez committed:
Fix blank append error

Steven-Baez committed:
Fix missing ItemProperties

Dariensg reviewed (changes requested):
Just that one comment on the Mixin switching to Inject if possible. Can discuss in Discord if you'd like

Steven-Baez commented:
yeah sure

ItsBlackGear committed:
platform implementation

- ported the whole mod to use builtin api instead of arch's api
- commented out revamped code.

ItsBlackGear committed:
my bad!

- summon bow item properties are back online!
- moved summon bow properties to post-client to avoid issues on forge side

Steven-Baez committed:
Fix empty name enchant bug

Steven-Baez committed:
New spawning event!

Steven-Baez committed:
Merge remote-tracking branch 'origin/summonentity' into summonentity

# Conflicts:
#	common/src/main/java/com/github/teamfusion/summonerscrolls/SummonerScrolls.java
#	common/src/main/resources/summonerscrolls-common.mixins.json

Dariensg reviewed (approved):
LGTM

Steven-Baez commented:
thanks!
